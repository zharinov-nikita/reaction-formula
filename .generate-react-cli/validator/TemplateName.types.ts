export interface ValidatorTemplateNameProps {}
export interface ValidatorTemplateName {}
