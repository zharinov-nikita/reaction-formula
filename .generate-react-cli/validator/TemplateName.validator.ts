import { ValidatorTemplateName, ValidatorTemplateNameProps } from './template-name.types'

export function validatorTemplateName(props: ValidatorTemplateNameProps): ValidatorTemplateName {
  return {}
}
