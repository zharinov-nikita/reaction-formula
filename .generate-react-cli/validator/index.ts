export * from './template-name.validator'
export * from './template-name.types'