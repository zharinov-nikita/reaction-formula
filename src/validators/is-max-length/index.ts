export * from './is-max-length.validator'
export * from './is-max-length.types'
