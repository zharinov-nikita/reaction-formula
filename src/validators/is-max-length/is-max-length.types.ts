export interface ValidatorIsMaxLengthProps {
  value: string
  maxLength: number
  clearOfSpaces?: boolean
}
export type ValidatorIsMaxLength = boolean
