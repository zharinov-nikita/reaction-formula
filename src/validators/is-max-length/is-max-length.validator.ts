import { ValidatorIsMaxLength, ValidatorIsMaxLengthProps } from './is-max-length.types'

export function validatorIsMaxLength(props: ValidatorIsMaxLengthProps): ValidatorIsMaxLength {
  if (!props.clearOfSpaces) return props.value.length > props.maxLength
  return props.value.replace(/\s/g, "").length > props.maxLength
}
