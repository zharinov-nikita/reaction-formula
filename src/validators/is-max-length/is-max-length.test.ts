import { validatorIsMaxLength } from './is-max-length.validator'

describe('validatorIsMaxLength', () => {
  test('expected true (with spaces)', () => {
    expect(validatorIsMaxLength({ value: 'hi ', maxLength: 2 })).toBe(true)
  })

  test('expected false (with spaces)', () => {
    expect(validatorIsMaxLength({ value: 'hi ', maxLength: 4 })).toBe(false)
  })

  test('expected true (clear of spaces)', () => {
    expect(validatorIsMaxLength({ value: 'hi ', maxLength: 1, clearOfSpaces: true })).toBe(true)
  })

  test('expected false (clear of spaces)', () => {
    expect(validatorIsMaxLength({ value: 'hi ', maxLength: 3, clearOfSpaces: true })).toBe(false)
  })

  test('snapshot', () => {
    expect(validatorIsMaxLength({ value: 'hi ', maxLength: 2 })).toMatchSnapshot()
    expect(validatorIsMaxLength({ value: 'hi ', maxLength: 4 })).toMatchSnapshot()
    expect(validatorIsMaxLength({ value: 'hi ', maxLength: 1, clearOfSpaces: true })).toMatchSnapshot()
    expect(validatorIsMaxLength({ value: 'hi ', maxLength: 3, clearOfSpaces: true })).toMatchSnapshot()
  })
})
