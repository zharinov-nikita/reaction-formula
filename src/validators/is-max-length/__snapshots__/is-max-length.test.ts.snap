// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validatorIsMaxLength snapshot 1`] = `true`;

exports[`validatorIsMaxLength snapshot 2`] = `false`;

exports[`validatorIsMaxLength snapshot 3`] = `true`;

exports[`validatorIsMaxLength snapshot 4`] = `false`;
